presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveParquetDereferencePushDown.java

```diff
@@ -138,7 +144,9 @@ private static Subfield createNestedColumn(RowExpression rowExpression, Expressi
         while (true) {
             if (rowExpression instanceof VariableReferenceExpression) {
                 Collections.reverse(elements);
-                return new Subfield(((VariableReferenceExpression) rowExpression).getName(), unmodifiableList(elements));
+                String name = ((VariableReferenceExpression) rowExpression).getName();
+                String originalColumnName = regularHiveColumnHandles.get(name).getName();
```

Comment:
Should we add check to make sure `regularHiveColumnHandles.get(name)` returns non-null?

Reply:
Added a Preconditions check